cli/app.go

```diff
@@ -486,6 +490,9 @@ func (c *App) runAppWithContext(command *kingpin.CmdClause, cb func(ctx context.
 		releasable.EnableTracking(releasable.ItemKind(r))
 	}
 
+	pproflogging.MaybeStartProfileBuffers(ctx)
+	defer pproflogging.MaybeStopProfileBuffers(ctx)
```

Comment:
on stop detach from the parent content
1. in case the parent context has errored out
2. to set a timeout for the dump